### PR TITLE
set default request timeout of 60 seconds, configurable

### PR DESCRIPTION
**Pull request by jeremyandrews**
- [#382](https://github.com/tag1consulting/goose/pull/382) set client timeout to 60 seconds by default, used for all requests made; introduce `--timeout VALUE` where VALUE is seconds as integer or a float; timeout can be configured programatically using `GooseDefault::Timeout`
 - introduces `utils::get_float_from_string()`, now used by `utils::get_hatch_rate()`
 - updates `set_client_builder()` documentation to reflect how this changes the default Reqwest Client configuration

### CODE REVIEW

File: src/util.rs [hunk 1]
```diff
@@ -355,18 +355,44 @@ pub fn ms_timer_expired(started: time::Instant, elapsed: usize) -> bool {
...
+/// assert_eq!(util::get_float_from_string(Some(".1".to_string())), Some(0.1));
+///
+/// // Valid float string returns a proper float.
+/// assert_eq!(util::get_float_from_string(Some("1.1".to_string())), Some(1.1));
+///
+/// // Invalid number with too many decimals returns None.
+/// assert_eq!(util::get_float_from_string(Some("1.1.1".to_string())), None);
+///
+/// // No number returns None.
+/// assert_eq!(util::get_float_from_string(None), None);
```
Comment (slashrsm) [inline src/util.rs:384]: I love doc tests!

**Review by slashrsm — COMMENTED** [review]
Looks good to my (n00b) me.

File: src/config.rs [hunk 2]
```diff
@@ -2046,6 +2084,17 @@ impl GooseConfiguration {
             }
         }
 
+        // If set, timeout must be non-zero.
+        if let Some(timeout) = self.timeout.as_ref() {
+            if timeout == "0" {
```
Comment (LionsAd) [inline src/config.rs:2089]: Should this not already do a test conversion to check if eg “abc” is passed?

Comment (jeremyandrews) [reply src/config.rs:2089]: It’s validate le already to be an unsigned integer.  Now we validate it’s not 0.

Comment (jeremyandrews) [reply src/config.rs:2089]: actually: I'm wrong -- we were validating in an earlier version but it's currently broken -- I'll fix.

Comment (jeremyandrews) [reply src/config.rs]: fixed so it's properly validating the timeout is > 0.0

**Review by LionsAd — APPROVED** [review]
RTBM

**Pull request merged**